### PR TITLE
-Fix QuestionnaireResponse.subject issue.

### PR DESCRIPTION
**Pull request by ahsanbhatti49**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] or Closes #[issue number]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [x] I have verified any strings visible on UI components are in the `strings.xml` file
- [x] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [x] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt [hunk 1]
```diff
@@ -205,7 +205,12 @@ constructor(
         val bundle = performExtraction(context, questionnaire, questionnaireResponse)
         bundle.entry.forEach { bundleEntry ->
           // add organization to entities representing individuals in registration questionnaire
-          if (bundleEntry.resource.resourceType.isIn(ResourceType.Patient, ResourceType.Group)) {
+          if (bundleEntry.resource.resourceType.isIn(
+              ResourceType.Patient,
+              ResourceType.Group,
+              ResourceType.Encounter
+            )
```
[the remark this answers is not in the document]
Comment (ahsanbhatti49) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:209]: @ellykits  In my case, the app was crashing everytime questionnaireResponse is generated that does not contain any subject. And questionnaireConfig.resourceIdentifier == null. so in this case I need to enable the check to fetch subject from extracted bundle like below.
in this check 
 // if it is new registration set response subject
            if (questionnaireConfig.resourceIdentifier == null)
              questionnaireResponse.subject = bundleEntry.resource.asReference()


cc : @f-odhiambo

[the remark this answers is not in the document]
Comment (f-odhiambo) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:209]: I am of the opinion we remove the check, tested and works as expected, I, however, am not sure what it was checking exactly, I guess trying to exclude other resources based on the comment 
//Add organization to entities representing individuals in the registration questionnaire

CC @ellykits @ahsanbhatti49

[the remark this answers is not in the document]
Comment (f-odhiambo) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:209]:
````
 // add organization to entities representing individuals in registration questionnaire
if (bundleEntry.resource.resourceType.isIn(ResourceType.Patient, ResourceType.Group,ResourceType.Encounter)) {
            // if it is new registration set response subject
          if (questionnaireConfig.resourceIdentifier == null)
            questionnaireResponse.subject = bundleEntry.resource.asReference()
         // }
````

[the remark this answers is not in the document]
Comment (f-odhiambo) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt:209]: Pushing an update the PR

**Comment by pld** [comment]
This failure is interesting, this looks legitimate, is this expected?

```
FhirCarePlanGeneratorTest > generateOrUpdateCarePlan should generate careplan for 5 visits when lmp has passed 3 months FAILED
    junit.framework.AssertionFailedError at FhirCarePlanGeneratorTest.kt:1800
```

**Comment by pld** [comment]
> This failure is interesting, this looks legitimate, is this expected?
> 
> ```
> FhirCarePlanGeneratorTest > generateOrUpdateCarePlan should generate careplan for 5 visits when lmp has passed 3 months FAILED
>     junit.framework.AssertionFailedError at FhirCarePlanGeneratorTest.kt:1800
> ```

oh nevermind seeing this on an unrelated PR, seems flakey

**Comment by pld** [comment]
> > This failure is interesting, this looks legitimate, is this expected?
> > ```
> > FhirCarePlanGeneratorTest > generateOrUpdateCarePlan should generate careplan for 5 visits when lmp has passed 3 months FAILED
> >     junit.framework.AssertionFailedError at FhirCarePlanGeneratorTest.kt:1800
> > ```
> 
> oh nevermind seeing this on an unrelated PR, seems flakey

and on main, will investigate...

**Review by pld — APPROVED** [review]

**Comment by pld** [comment]
#2402 handles follow-up to fix the failing test

**Pull request merged**